portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppLocalServiceImpl.java

```diff
@@ -710,6 +712,18 @@ public FileEntry getFileEntryByUuidAndGroupId(String uuid, long groupId)
 			groupId);
 
 		for (Repository repository : repositories) {
+
+			if (Objects.equals(
+				repository.getClassName(),
+				TemporaryFileEntryRepository.class.getName())) {
+
+				if (_log.isDebugEnabled()) {
+					_log.debug("Skipping temporary file entry repository");
+				}
+
+				continue;
+			}
+
```

Comment:
Instead of checking class name, can you check for the presence of the `TemporaryFileEntriesCapability`? This would make it skip any temporary repository (not only that particular one).